libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/tripsession/TripSessionStarterViewModel.kt

```diff
@@ -0,0 +1,82 @@
+package com.mapbox.navigation.dropin.component.tripsession
+
+import android.annotation.SuppressLint
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.dropin.component.navigation.NavigationState
+import com.mapbox.navigation.dropin.component.navigation.NavigationStateViewModel
+import com.mapbox.navigation.dropin.lifecycle.UIViewModel
+import com.mapbox.navigation.utils.internal.logI
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.launch
+
+sealed class TripSessionStarterAction {
+    object EnableTripSession : TripSessionStarterAction()
+    object EnableReplayTripSession : TripSessionStarterAction()
+}
+
+data class TripSessionStarterState(
+    // TODO this is true for development. Road testing should be set to false.
+    val isReplayEnabled: Boolean = true,
+)
+
+@OptIn(ExperimentalPreviewMapboxNavigationAPI::class)
+@SuppressLint("MissingPermission")
+class TripSessionStarterViewModel(
```

Comment:
Probably `TripSessionViewModel`

Reply:
That was my initial attempt. But there are lots of conflicts with core-nav `TripSessionState`

I was needing to add the package `com.mapbox.navigation.core.trip.session.TripSessionState` to distinguish them

Reply:
Okay makes sense. 👍🏻